### PR TITLE
:sparkles: Support multi-container Providers.

### PR DESCRIPTION
**Pull request by jortel**
ref: https://github.com/konveyor/tackle2-hub/issues/599

### CODE REVIEW

File: builder/deps.go [hunk 1]
```diff
@@ -59,6 +59,9 @@ func (b *Deps) read() (input []output.DepsFlatItem, err error) {
 	input = []output.DepsFlatItem{}
 	f, err := os.Open(b.Path)
 	if err != nil {
+		if os.IsNotExist(err) {
```
Comment (jortel) [inline builder/deps.go:62]: **hint**: some providers do not report deps.

File: cmd/analyzer.go [hunk 2]
```diff
@@ -67,7 +67,11 @@ func (r *Analyzer) options(output string) (options command.Options, err error) {
 	if err != nil {
 		return
 	}
-	settings.Report()
+	f, err := addon.File.Post(settings.path())
+	if err != nil {
+		return
+	}
+	addon.Attach(f)
```
Comment (jortel) [inline cmd/analyzer.go:74]: **hint**: Attach settings for better reporting.

File: cmd/settings.go [hunk 3]
```diff
@@ -27,7 +30,32 @@ func (r *Settings) Read() (err error) {
...
+	err = yaml.Unmarshal(b, &r.content)
+	if err != nil {
+		return
+	}
+	r.index = len(r.content)
+	err = r.AppendExtensions()
+	return
+}
+
+// AppendExtensions adds extension fragments.
```
Comment (jortel) [inline cmd/settings.go:42]: **hint**: main magic here for aggregating the provider init configs.

File: cmd/settings.go [hunk 4]
```diff
@@ -101,20 +121,18 @@ func (r *Settings) ProxySettings() (err error) {
...
+	if len(http)+len(https) == 0 {
+		return
+	}
+	extensions := r.content[r.index:]
+	for i := range extensions {
+		p := &extensions[i]
+		p.Proxy = &provider.Proxy{
+			HTTPProxy:  http,
+			HTTPSProxy: https,
+			NoProxy: strings.Join(
```
Comment (jortel) [inline cmd/settings.go:133]: **hint**: newer method for passing proxies to providers.

File: hack/addon.yaml [hunk 5]
```diff
@@ -5,5 +5,100 @@ metadata:
...
+      initConfig:
+
+---
+kind: Task
+apiVersion: tackle.konveyor.io/v1alpha1
+metadata:
+  namespace: konveyor-tackle
+  name: analyzer
+spec:
+  priority: 10
```
Comment (jortel) [inline hack/addon.yaml:103]: **hint**: The default priority is 0. priority 10 here assumes discovery tasks will be 0 and all user initiated tasks would be 10 leaving a little room for priorities in between.

**Review by mansam — APPROVED** [review]
LGTM, pending the go.mod fixup

**Pull request merged**